line_profiler/__init__.py

```diff
@@ -13,10 +14,11 @@
 
 from .line_profiler import __version__
 
-from .line_profiler import (LineProfiler, LineProfilerMagics,
-                            load_ipython_extension, load_stats, main,
+from .line_profiler import (LineProfiler, load_stats, main,
                             show_func, show_text,)
 
+from .ipython_extension import load_ipython_extension, LineProfilerMagics
```

Comment:
This won't work with lazy import, at this point ``LineProfilerMagics`` is always ``None``, and it won't change when the value is updated in ``ipython_extension``.

Reply:
Will it be the case that we only care if `load_ipython_extension` is defined if `IPython` is in `sys.modules`? That could be a potential optimization that prevents the large import in cases where it is not needed.

Reply:
Good idea, it will be cleaner this way and should work with you other suggestion below.